### ISSUE TITLE
Search does not include the forums

### ISSUE THREAD

**Issue by AdamBark** [tendenci/tendenci#824]
Is it possible to index the forums and make them searchable?

**Comment by eschipul** [comment]
Yes, but beware of spammers. I got so tired of them that our forums are blocked in robots.txt.

If you want, submit a pull request with the forums index fields you want that first checks if the module is enabled. 

Yet again, I must warn you that associations generate a ton of unique fresh content and are therefore major targets for bad guys. If you allow self reg without approval you may find yourself spending hours deleting posts and users.

On the plus side, if you limit to members and tweak your nginx config, and maybe .htaccess as well, some clients have a vibrant community in forums.

Lastly, don’t forget you can pull down a template in the theme editor and add conversations below any object (articles, pages, news, events, etc...) via disqus https://www.tendenci.com/works-with/

**Comment by AdamBark** [comment]
So you discourage the use of the forums altogether, even though they're enabled by default? I had it running for a while before without any spam but I disabled it because it wasn't being used but we have a new use case now.

**Comment by eschipul** [comment]
Forums are awesome. Just easily abused if not actively monitored. But they are a great community building feature. Internally we use forums for our stand-up / agile reports, they are just set to private so only our team can see them.

Another helpful module is “tickets” or “helpdesk” which we use to track issues with paid hosting clients on RP (recurring payments.)

It just depends on your use case. But we’re definitely open to a pull request to index forums. If that is “programmer speak” I can amend this issue with greater detail for you? I’d be happy to do that.

If you go to demo.tendenci.com/explorer/ and click “play” and then “show schema” - then search for forums - you can see the db structure if you don’t have a running version. (Just remember demo site resets every two hours - again because of spammers.)

**Comment by bernd-wechner** [comment]
Forums are typically a waste of time in my experience, not least because of issue of maintenance and that far more focussed service providers cover every niche you can think of. Want support or to provide it, use stackoverflow, or one of it's many sites using the same tech (to keep quality high and abuse down). And then there are all the social media tools ... aargh we are already swimming in fora ...

That said, the fact that Tendenci forums don't have an internal search takes them from as good as useless - but I had to use them to look for support: https://www.tendenci.com/forums/c/general/feature-requests/forum-search/ to abjectly useless as stated therein more politely. As I can't even search the support forum to see if someone's suggested that feature already.

And no it shouldn't part of site wide search unless that has a filter that can restrict it to forum searches. 

As to abuse on forums, sure, but this association management software and I don't expect the public to be able to create an account and if we did, surely a captcha, an email confirmation and as a last step for true quality, human approval required before a join is finished. 

Anyhow, I mean no disrespect, but the only Tendenci forum I ever expect to use is the Tendenci support forum and that is difficult to sue without search clearly. Any forum is.

**Comment by bernd-wechner** [comment]
@eschipul Awesome to hear from you hear:

https://www.tendenci.com/forums/c/general/feature-requests/forum-search/

of course the need for search in forums is already here on GitHub, and here's the item to track implementation of search in forums. Indeed, a low priority as are forums full stop because of the problem of spammers. Forums are in 2021 IMHO near useless, and best left to the big players with good spam control (and I'm thinking for support that includes GitHub Issues and the StackOverflow family of sites).

But as noted here forums with a search facility are less than near useless (abjectly useless) as it's not even possible to search for answers prior to posting and hence the same questions are doomed to be asked again and again, as one of the first QA requests from most forum sites is: before you post search to see if your question is already answered. 

If the tendenci forums: https://www.tendenci.com/forums

Are to be useful, the need a search facility. And if they are not useful, why not just disable the forums app altogether on the tendenci.com site? Just a suggestion. For example in Oct 2020 I posted a few questions there and a) I had not way to check if they'd been asked already, and b) it took until Feb 2021 (so 4 months) before anyone answered ... all of which suggests, fairly low utility of the forums there.
